source/common/common/fancy_logger.cc

```diff
@@ -82,6 +82,17 @@ std::string FancyContext::listFancyLoggers() ABSL_LOCKS_EXCLUDED(fancy_log_lock_
   return info;
 }
 
+absl::flat_hash_set<absl::string_view> FancyContext::getFancyLoggerKeys()
```

Comment:
you don't need to make this new map to avoid the n^2 issue. You just need to use the map you already have for lookups rather than for iteration.